spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/GCPProperties.java

```diff
@@ -27,7 +28,7 @@
 
 	private String projectId;
 
-	private String jsonKeyFile;
+	private Resource credentialKeyFile;
```

Comment:
Tha analogy is AWS. There's an AWSCredentials object that gets created from SECRET_KEY and ACCESS_KEY. Those two strings are represented in a Properties class. The auto configuration creates the AWSCredentials using those two provided properties.

Could users use the values from the properties to get the credentials? Yes, if they are willing to go and create a new Credentials object, but why would they if there's already a bean created for them.

Same applies for a datasource for example. you have a property called `datasource.username` & `datasource.password` that is available for anyone via properties, but Boot will create a DataSource object for you so you don't need to create yourself